hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/response/lifecycle/TestOMLifecycleConfigurationCreateResponse.java

```diff
@@ -0,0 +1,159 @@
+    OzoneManager ozoneManager = mock(OzoneManager.class);
+    OzoneConfiguration ozoneConfiguration = new OzoneConfiguration();
+    ozoneConfiguration.set(OMConfigKeys.OZONE_OM_DB_DIRS,
+        tempDir.getAbsolutePath());
+    omMetadataManager = new OmMetadataManagerImpl(ozoneConfiguration, ozoneManager);
+    batchOperation = omMetadataManager.getStore().initBatchOperation();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    if (batchOperation != null) {
+      batchOperation.close();
+    }
+  }
+
+  @Test
+  public void testAddToDBBatch() throws Exception {
+    String volumeName = UUID.randomUUID().toString();
+    String bucketName = UUID.randomUUID().toString();
+
+    OmLifecycleConfiguration omLifecycleConfiguration =
+        new OmLifecycleConfiguration.Builder()
+            .setVolume(volumeName)
+            .setBucket(bucketName)
+            .setBucketLayout(BucketLayout.OBJECT_STORE)
+            .addRule(new OmLCRule.Builder().setPrefix("")
+                .addAction(new OmLCExpiration.Builder().setDays(30).build())
+                .build())
+            .setCreationTime(Time.now())
+            .build();
+
+    assertEquals(0, omMetadataManager.countRowsInTable(
+        omMetadataManager.getLifecycleConfigurationTable()));
+
+    CreateLifecycleConfigurationResponse createLifecycleConfigurationResponse =
+        CreateLifecycleConfigurationResponse.newBuilder()
+            .build();
+
+    OMLifecycleConfigurationCreateResponse response =
+        new OMLifecycleConfigurationCreateResponse(OMResponse.newBuilder()
+            .setCmdType(Type.CreateLifecycleConfiguration)
+            .setStatus(Status.OK)
+            .setCreateLifecycleConfigurationResponse(
+                createLifecycleConfigurationResponse).build(),
+            omLifecycleConfiguration);
+
+    response.addToDBBatch(omMetadataManager, batchOperation);
+
+    // Do manual commit and see whether addToBatch is successful or not.
+    omMetadataManager.getStore().commitBatchOperation(batchOperation);
+
+    assertEquals(1, omMetadataManager.countRowsInTable(
+        omMetadataManager.getLifecycleConfigurationTable()));
+
+    Table.KeyValue<String, OmLifecycleConfiguration> keyValue =
+        omMetadataManager.getLifecycleConfigurationTable()
+            .iterator()
+            .next();
+
+    // Lifecycle configuration keys follow bucket key format.
+    assertEquals(omMetadataManager.getBucketKey(volumeName, bucketName),
+        keyValue.getKey());
+
+    assertTrue(isSame(omLifecycleConfiguration, keyValue.getValue()));
+  }
+
+  /**
+   * Compares two lifecyce and returns false if they are not identical, as
+   * the equal and hash methods are not overridden in OmLifecycleConfiguration.
+   *
+   * @param o1 first lifecycle configuration
+   * @param o2 second lifecycle configuration
+   * @return true if both are identical false otherwise
+   */
+  private boolean isSame(OmLifecycleConfiguration o1,
```

Comment:
Can we define an equals() function in OmLifecycleConfiguration, and use equals() to do the compare?

Reply:
We can compare the protobuf value too, this is a easy way.

Reply:
This change is missed.